components/tdisplays3/display.py

```diff
@@ -28,10 +28,45 @@ def validate_tdisplays3(config):
 )
 
 async def to_code(config):
+    # Add platformio build_flags for the correct TFT_eSPI settings for the T-Display-S3
+    # This allows using current, unpatched versions of TFT_eSPI
+    cg.add_build_flag("-DUSER_SETUP_LOADED=1")
+    cg.add_build_flag("-DST7789_DRIVER=1")
+    cg.add_build_flag("-DINIT_SEQUENCE_3=1")
+    cg.add_build_flag("-DCGRAM_OFFSET")
+    cg.add_build_flag("-DTFT_RGB_ORDER=TFT_RGB")
+    cg.add_build_flag("-DTFT_INVERSION_ON=1")
+    cg.add_build_flag("-DTFT_PARALLEL_8_BIT=1")
+    cg.add_build_flag("-DTFT_WIDTH=170")
+    cg.add_build_flag("-DTFT_HEIGHT=320")
+    cg.add_build_flag("-DTFT_DC=7")
+    cg.add_build_flag("-DTFT_RST=5")
+    cg.add_build_flag("-DTFT_WR=8")
+    cg.add_build_flag("-DTFT_RD=9")
+    cg.add_build_flag("-DTFT_D0=39")
+    cg.add_build_flag("-DTFT_D1=40")
+    cg.add_build_flag("-DTFT_D2=41")
+    cg.add_build_flag("-DTFT_D3=42")
+    cg.add_build_flag("-DTFT_D4=45")
+    cg.add_build_flag("-DTFT_D5=46")
+    cg.add_build_flag("-DTFT_D6=47")
+    cg.add_build_flag("-DTFT_D7=48")
+    cg.add_build_flag("-DLOAD_GLCD=1")
+    cg.add_build_flag("-DLOAD_FONT2=1")
+    cg.add_build_flag("-DLOAD_FONT4=1")
+    cg.add_build_flag("-DLOAD_FONT6=1")
+    cg.add_build_flag("-DLOAD_FONT7=1")
+    cg.add_build_flag("-DLOAD_FONT8=1")
+    cg.add_build_flag("-DLOAD_GFXFF=1")
+    cg.add_build_flag("-DSMOOTH_FONT=1")
```

Comment:
Those defines seem related to fonts; which shouldn't be relevant when using ESPHome fonts. Could you test with your setup, whether it works without those defines?
```suggestion

```